example-apps/token_curated_registry/db/mapper.go

```diff
@@ -0,0 +1,161 @@
+package db
+
+import (
+	"github.com/cosmos/cosmos-academy/example-apps/token_curated_registry/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/tendermint/go-amino"
+)
+
+type BallotMapper struct {
+	ListingKey sdk.StoreKey
+
+	CommitKey sdk.StoreKey
+
+	RevealKey sdk.StoreKey
+
+	BallotKey sdk.StoreKey
+
+	Cdc *amino.Codec
+}
+
+func NewBallotMapper(listingKey sdk.StoreKey, ballotkey sdk.StoreKey, commitKey sdk.StoreKey, revealKey sdk.StoreKey, _cdc *amino.Codec) BallotMapper {
+	return BallotMapper{
+		ListingKey: listingKey,
+		CommitKey:  commitKey,
+		RevealKey:  revealKey,
+		BallotKey:  ballotkey,
+		Cdc:        _cdc,
+	}
+}
+
+// Will get Ballot using unique identifier. Do not need to specify status
+func (bm BallotMapper) GetBallot(ctx sdk.Context, identifier string) types.Ballot {
+	store := ctx.KVStore(bm.BallotKey)
+	key := []byte(identifier)
+	val := store.Get(key)
+	if val == nil {
+		return types.Ballot{}
+	}
+	ballot := &types.Ballot{}
+	err := bm.Cdc.UnmarshalBinary(val, ballot)
+	if err != nil {
+		panic(err)
+	}
+	return *ballot
+}
+
+func (bm BallotMapper) AddBallot(ctx sdk.Context, identifier string, owner sdk.Address, applyLen int64, bond int64) sdk.Error {
+	store := ctx.KVStore(bm.BallotKey)
+
+	newBallot := types.Ballot{
```

Comment:
I think you should create instead a `NewBallot(<args>) trc.Ballot` function here instead of creating the struct right away